### PR TITLE
add draft ADR for static IP support

### PR DESCRIPTION
**Pull request by bengerman13**
## Changes proposed in this pull request:
- add draft ADR for static IP support

## security considerations
None

### CODE REVIEW

**Comment by davemcorwin** [comment]
Thanks @bengerman13 Just for clarification, I agree with the guidance that the canonical domain should be `www` and the apex should redirect to it, that said, does the implementation suggested here support serving the site at the apex with Cloudfront if that is what the user wants?

**Comment by bengerman13** [comment]
not any way that is not already supported (e.g. aliases and bypassing our recommendation to use the intermediate external-domains-production.cloud.gov address)
Static addressing and cloudfront are totally incompatible. The only way to accomplish it would be by proxying to cloudfront, which defeats the point of cloudfront.

File: adr/ADR-0003-static-ips.md [hunk 1]
```diff
@@ -23,6 +23,17 @@ We should support static IP addressing for custom domains on ALBs. We'll do this
...
 
+The user process for using this looks like:
+1. Decide on a hostname (in our example, myapp.agency.gov)
+1. Create the cf domain 
+   `cf create-private-domain myapp.agency.gov`
+1. Push an app with that domain as a route
+1. Create the domain service instance
+   `cf create-service external-domain domain my-domain -p '{"domains": ["myapp.agency.gov"]}'`
+1. Wait for domain to finish provisioning
+1. (exact process to be determined) interact with CF to get the associated static IPs
```
Comment (davemcorwin) [inline adr/ADR-0003-static-ips.md:34]: Won't the same static IPs be used for everyone? Perhaps we can just add this in the documentation and/or in the output text from the broker?

Comment (bengerman13) [reply adr/ADR-0003-static-ips.md:34]: we have several ALBs in production, and there will be one set of IPs per ALB

**Comment by mheadd** [comment]
Is figuring out a way to emit the static IP info when the domain is set up the3 first order of business here? Is this feasible if we determine at some point that there is not an easy way to do that?

**Comment by bengerman13** [comment]
There are definitely ways to do it, so I think we can barrel forward without a firm plan on that.

The ideal is using the [metadata field](https://github.com/openservicebrokerapi/servicebroker/blob/v2.16/spec.md?plain=1#L1018) from the Open Service Broker API. I have a vague memory of that not being well-supported by CAPI, but I could be wrong, or it could have changed.

As another option, we could emit it as part of the [last operation](https://github.com/openservicebrokerapi/servicebroker/blob/v2.16/spec.md?plain=1#L727) data, which is what the old broker did for ALB names (there are downsides there - namely, we need to make sure it's emitted for any Last Operation step)

Finally, if neither of those work, we could expose this via service keys - this seems like the clumsiest user experience IMO, but maybe the closest to what CAPI/CF want us to do

**Review by mheadd — APPROVED** [review]
🚀

**Pull request merged**